### PR TITLE
New subworkflow: fastq_align_bwaaln

### PR DESCRIPTION
**Pull request by jfy133**
Implements the bwa aln -> samse/sampe -> index workflow.

I've not modified the excisitng `fastq_align_bwa` to maybe the more 'correct' `fastq_align_bwamem`, but maybe this is something to consider @JoseEspinosa? I'm wondering if that should also be in guidelines (full module name, if subcommands exist?)

Note that _unlike_ other alignment subworkflows, I've explicitly _not_ included the stats here. I'm wary of doing this at this step given in some cases people (i.e., me) may actually want to map per run/lane/library separately, then merge the resulting BAM files after this (of a given sample/library), upon which you generte stats (so to get stats per library or sample rather than of each lane/run separately). This helps chunk up alignment steps of pipelines into smaller/'cluster-friendly' batches of jobs and often will run faster (than one monolith - something we currently do in eager DSL1 and people complained about).

Therefore, I also wonder if this could be another guideline... 'stats' related subworkflows should be separate from 'functional' workflows?  In sense we don't want to assume too 'hard' what a pipeline developer wants to generate stats from? Although this would maybe require updating of previous subworkflows

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: subworkflows/nf-core/fastq_align_bwaaln/main.nf
Comment (nvnieuwk) [inline subworkflows/nf-core/fastq_align_bwaaln/main.nf:13]:
```suggestion
    ch_reads // channel (mandatory): [ val(meta), [ reads ] ]. subworkImportant: meta REQUIRES single_end` entry!
```

File: subworkflows/nf-core/fastq_align_bwaaln/meta.yml
Comment (nvnieuwk) [inline subworkflows/nf-core/fastq_align_bwaaln/meta.yml:26]:
```suggestion
        A directory containing bwa aln reference indices as from `bwa index`
```

File: subworkflows/nf-core/fastq_align_bwaaln/meta.yml
Comment (nvnieuwk) [inline subworkflows/nf-core/fastq_align_bwaaln/meta.yml:29]: Those are not index extensions ;)

File: tests/subworkflows/nf-core/fastq_align_bwaaln/main.nf
Comment (nvnieuwk) [inline tests/subworkflows/nf-core/fastq_align_bwaaln/main.nf:19]:
```suggestion
```

**Review by nvnieuwk — APPROVED** [review]
Some typos but overall LGTM! (But you'll better get a second opinion as alignment is not my strong suite :grin: )

**Review by muffato — COMMENTED** [review]
I approve too, pending the suggestions from @nvnieuwk

File: subworkflows/nf-core/fastq_align_bwaaln/meta.yml
Comment (jfy133) [inline subworkflows/nf-core/fastq_align_bwaaln/meta.yml:29]:
```suggestion
```

**Review by JoseEspinosa — APPROVED** [review]
LGTM! :rocket: 
Regarding the naming of the workflow I agree that we should change it. For the stats I tend to agree but would like to see what others maintainers think.

**Comment by jfy133** [comment]
Ok, will fix linting then merge for now as you've brought up the discussion :)

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Pull request merged**
